### PR TITLE
hqemu: init at 2.5.2

### PR DESCRIPTION
**Pull request by risicle**
###### Motivation for this change
hqemu (http://csl.iis.sinica.edu.tw/hqemu/) is an interesting research project that grafts LLVM on to qemu's TCG binary translation, allowing it to optimize hotspots. I've managed to get some pretty good results from it, tripling execution speeds in some of my tests.

Package is heavily based on the qemu one, with some features removed and disabled for darwin - I get the feeling linux is the only host that has ever seen any attention, and my brief attempts at getting it to build on macos yielded nothing but pain.

Also note it has `knownVulnerabilities` set because it's not as though this is a security-maintained application, so isn't really appropriate for production use.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Review by doronbehar — APPROVED** [review]

**Comment by typetetris** [comment]
Looks like a qemu derivative work, which would be GPL licensed. Don't know if the content of `COPYRIGHT` in the tarball is compatible with GPL.

**Comment by risicle** [comment]
That's a very good point - hadn't noticed that. Do we have a "license: unclear"?

**Comment by typetetris** [comment]
I don't think there is `license: unclear`. Maybe try to get the upstream authors to use something more clear?

**Comment by risicle** [comment]
The upstream project is basically an academic paper and it doesn't seem to really be maintained anymore. The license sounds like one of those "university ip policy" disclaimers and I don't see much hope in getting clarity from "upstream".

**Comment by prusnak** [comment]
@risicle do you still want to get this in?

**Comment by risicle** [comment]
Well, the latest status is I'm unclear hqemu's license

**Comment by doronbehar** [comment]
I think it'd be OK to add this package to NixOS, with the `unfree` license, if upstream doesn't say differently.

File: pkgs/applications/virtualization/hqemu/default.nix [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+, virglSupport ? openGLSupport, virglrenderer
+, smbdSupport ? false, samba
+, hostCpuOnly ? false
+, hostCpuTargets ? (if hostCpuOnly
+                    then (stdenv.lib.optional stdenv.isx86_64 "i386-softmmu"
+                          ++ ["${stdenv.hostPlatform.qemuArch}-softmmu"])
+                    else null)
+}:
+
+with stdenv.lib;
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/hqemu/default.nix:25]:
```suggestion
with lib;
```

File: pkgs/applications/virtualization/hqemu/default.nix [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                  $out/bin/qemu-kvm \
+                  --add-flags "\$([ -e /dev/kvm ] && echo -enable-kvm)"
+    fi
+  '';
+
+  passthru = {
+    qemu-system-i386 = "bin/qemu-system-i386";
+  };
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/hqemu/default.nix:148]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
@ofborg eval

**Comment by risicle** [comment]
TBH I think I'm going to close this PR because I don't want to have to maintain this, or even keep its list of known vulnerabilities up to date. I won't delete the branch, because this at least serves a purpose as a piece of software archaeology - a project that's tricky to build that at least now has a "known good" point in time that it can be reliably built.

**Comment by Mic92** [comment]
Closing because of lack of activity from the author. Please re-open if you plan to work on this.

**Pull request closed without merging**